src/Components/Controller.h

```diff
@@ -0,0 +1,74 @@
+// MIT License
+
+// Copyright (c) 2021 YeiTingHao ShiChengFeng
+
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+
+// The above copyright notice and this permission notice shall be included in all
+// copies or substantial portions of the Software.
+
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+// SOFTWARE.
+
+#ifndef STREAM_CONTROLLER_H_
+#define STREAM_CONTROLLER_H_
+
+#include <string>
+
+class Render;
+class Decoder;
+class Router;
+class Encoder;
+class Capture;
+class DisplayGLWidget;
+
+// A class to control the streaming status
+// Example:
+// stream_controller = new Controller();
+// // init
+// stream_controller->init(ui.displayGLWidget);
+// // start live
+// stream_controller->live(port);
+// // start watch
+// stream_controller->watch(ip, listen_port, stream_port);
+// // leave
+// stream_controller->leave();
+
+class Controller {
+public:
+    // Default constructor
+    Controller() = default;
+    // Default destructor
+    ~Controller() = default;
+    // Set display widget pointer
+    void init(DisplayGLWidget*);
+    // Start streaming to target port 
+    void live(const int);
+    // Start receiving packet for source_ip:source_port and streaming ro output_port
+    void watch(const std::string&, const int, const int);
```

Comment:
Please add parameter naming to the definition, so people know which is source_port and which is output_port.